### DISCUSSION TITLE
Layer styles

### DISCUSSION THREAD

**Discussion by jllucero** [opengeos/leafmap#47] [Q&A] [answered]
Hello.  I'd like help on changing the colors, line thickness, and fill for layers.  I tried the "style" option from ipyleaflets for the folium plotting backend.  But it doesn't work.  I'd appreciate some examples.

Thank you!

**Comment by giswqs** [comment]
In folium, the style parameter name is called `style_function`. See below. 

https://python-visualization.github.io/folium/quickstart.html#Styling-function

![image](https://user-images.githubusercontent.com/5016453/122473560-07523380-cf90-11eb-8063-cf60414a5b73.png)

**Reply by jllucero** [reply]
Will this work if my layers were generated by "Map.add_gdf" and "Map.add_kml"?

**Reply by giswqs** [reply]
Yes, I believe so. All these functions have the kwargs argument, which will be passed to the add_geojson function behind the scene

**Reply by jllucero** [reply]
I'm trying to generalize that code to....
style_function = {"fillOpacity": 0.5,"weight": 0,"fillColor": "#black"}

leafmap.folium.TopoJson(
    Fire_Zones,
    "Fire Zones",
    style_function=style_function,
).add_to(Map)

But I get an error.

**Comment by giswqs** [comment, marked as answer]
See the example below 

```python
import leafmap.foliumap as leafmap

Map = leafmap.Map(center=[0, 0], zoom=2)

url = "https://raw.githubusercontent.com/giswqs/leafmap/master/examples/data/countries.geojson"

def style(feature): 
    return {
    "stroke": True,
    "color": "#0000ff",
    "weight": 2,
    "opacity": 1,
    "fill": True,
    "fillColor": "#0000ff",
    "fillOpacity": 0.1,
}

Map.add_geojson(url, layer_name="Countries", style_function=style)
Map
```

![](https://i.imgur.com/OAUNIwR.png)

**Reply by jllucero** [reply]
Thank you very much!  It works!

**Comment by giswqs** [comment]
I just improved support for folium layer styles. Now the same code will work for both ipyleaflet and folium without making any changes. Check out the updated notebook example https://leafmap.org/notebooks/10_add_vector, which uses the folium plotting backend. You should get the same results using the ipyleaflet plotting backend.

**Reply by jllucero** [reply]
Thank you!  This looks great.

**Comment by javieramuy** [comment]
Hello, good evening
is it a way to personalize the style from a csv or geojson multipoint layer. I tried many ways and i didn't figure it out. ChatGPT was helping me.

thanks in advance
